src/ixdat/readers/zilien.py

```diff
@@ -162,150 +177,314 @@ def read(self, path_to_file, cls=ECMSMeasurement, name=None, **kwargs):
+
+        Returns:
+            list: A list of Ixdat series objects.
+        """
+
+        series_objects = []
+        time_series = None
+
+        for column_number, column_header in enumerate(column_headers):
+            # Skip meta columns in the EC-lab dataset
+            if column_header in ("experiment_number", "technique_number"):
+                continue
+
+            column_data = data_rows_cut[:, column_number]
+
+            # Skip holes in the EC-lab dataset
+            if np.isnan(column_data).all():
+                continue
+
+            # Form series kwargs
+            series_name, unit, standard_name = names_and_units[column_number]
             series_kwargs = {
                 "name": series_name,
                 "unit_name": unit,
                 "data": column_data,
             }
-            if column_header == "Time [s]":  # Form TimeSeries
-                column_series = TimeSeries(**series_kwargs, tstamp=timestamp)
-                last_time_series = column_series
+
+            # Create the series
+            if column_header in ("Time [s]", "time/s"):
+                series_object = TimeSeries(**series_kwargs, tstamp=self._timestamp)
+                time_series = series_object
             else:
-                column_series = ValueSeries(**series_kwargs, tseries=last_time_series)
+                if time_series is None:
+                    raise ValueError("Time column must be first in a dataset series.")
 
-            series.append(column_series)
+                series_object = ValueSeries(**series_kwargs, tseries=time_series)
 
-        return series, aliases
+            series_objects.append(series_object)
+
+        return series_objects
+
+    # --- UTILS ---
+
+    @staticmethod
+    def _get_series_splits(series_headers):
+        """Create series names and their index pairs in the whole read dataset.
+
+        Args:
+            series_headers (list): Series name headers (even empty).
+
+        Returns:
+            list, list: List of tuples with index pairs
+            and a list with non-empty series name headers.
+        """
+        series_cut_indices = []
+        nonempty_headers = []
+
+        for index, series_header in enumerate(series_headers):
+            if series_header != "":
+                series_cut_indices.append(index)
+                nonempty_headers.append(series_header)
+
+        # create cut pairs
+        # zip_longest to have the last pair with the index of the last
+        # series header to the end
+        series_cut_indices = [
+            (begin, end)
+            for begin, end in zip_longest(series_cut_indices, series_cut_indices[1:])
+        ]
+
+        return series_cut_indices, nonempty_headers
+
+    @staticmethod
+    def _get_biologic_splits(technique_numbers):
+        """Create index pairs of row splits in the biologic dataset columns.
+
+        Args:
+            technique_numbers (np.array): Numbers of techniques during
```

Comment:
I can't use that, because no `Measurement` exists yet. It might return the same numbers, but it is just a different solution. Also, I don't want to create series in the background and confuse a user with not consecutive ids.

Reply:
good point.